uaa/src/test/java/org/cloudfoundry/identity/uaa/integration/ZoneControllerIntegrationTests.java

```diff
@@ -0,0 +1,160 @@
+import org.springframework.web.client.RestTemplate;
+
+@OAuth2ContextConfiguration(ZoneControllerIntegrationTests.ZoneClient.class)
+public class ZoneControllerIntegrationTests {
+
+    @Rule
+    public ServerRunning serverRunning = ServerRunning.isRunning();
+
+    private final UaaTestAccounts testAccounts = UaaTestAccounts.standard(serverRunning);
+
+    @Rule
+    public OAuth2ContextSetup context = OAuth2ContextSetup.standard(serverRunning);
+
+    @Rule
+    public TestAccountSetup testAccountSetup = TestAccountSetup.standard(serverRunning, testAccounts);
+
+    private RestTemplate client;
+
+    @Before
+    public void createRestTemplate() {
+        client = (OAuth2RestTemplate) serverRunning.getRestTemplate();
+        client.setErrorHandler(new OAuth2ErrorHandler(context.getResource()) {
+            // Pass errors through in response entity for status code analysis
+            @Override
+            public boolean hasError(ClientHttpResponse response) {
+                return false;
+            }
+
+            @Override
+            public void handleError(ClientHttpResponse response) {
+            }
+        });
+    }
+
+    @Test
+    public void testGetZone() {
+        //TODO: delete once the orchestrator create API implemented
+        IdentityZone identityZone = createZone();
+        assertNotNull(identityZone);
+        ResponseEntity<ZoneResponse> response = client.getForEntity(
+            serverRunning.getUrl("/zones") + "?name=" + identityZone.getName(),
+            ZoneResponse.class);
+        if (response.getStatusCode().is2xxSuccessful()) {
+            ZoneResponse zoneResponse = response.getBody();
+            assertNotNull(zoneResponse);
+            assertNotNull(identityZone);
+            assertNotNull(zoneResponse.getParameters());
+            assertEquals(identityZone.getSubdomain(), zoneResponse.getParameters().getSubdomain());
+            assertEquals(identityZone.getSubdomain(), zoneResponse.getConnectionDetails().getSubdomain());
+            String uri = "http://" + identityZone.getSubdomain() + ".localhost:8080/uaa";
+            assertEquals(uri, zoneResponse.getConnectionDetails().getUri());
+            assertEquals("http://localhost:8080/dashboard", zoneResponse.getConnectionDetails().getDashboardUri());
+            assertEquals(uri + "/oauth/token", zoneResponse.getConnectionDetails().getIssuerId());
+            assertEquals(X_IDENTITY_ZONE_ID, zoneResponse.getConnectionDetails().getZone().getHttpHeaderName());
+            assertEquals(identityZone.getId(), zoneResponse.getConnectionDetails().getZone().getHttpHeaderValue());
+        }
+    }
+
+    @Test
+    public void testGetZone_Notfound() {
+        ResponseEntity<String> response = client.getForEntity(
+            serverRunning.getUrl("/zones") + "?name=random-name",
+            String.class);
+        if (response.getStatusCode().is4xxClientError()) {
```

Comment:
Please remove if condition from here also like we did for mock movc tests.

Reply:
I think Mockmvc we are checking the status while fetch the request itself (andExpect(status().isNotFound())), but here we are not checking. So I use this condition to check.